plugins/askrene/askrene.c

```diff
@@ -162,7 +141,7 @@ static struct command_result *param_reserve_path(struct command *cmd,
 	if (tok->type != JSMN_ARRAY)
 		return command_fail_badparam(cmd, name, buffer, tok, "should be an array");
 
-	*path = tal(cmd, struct reserve_path);
+	*path = tal_arr(cmd, struct reserve_path, tok->size);
```

Comment:
No it wasn't meant to be an array. It is a structure with an array of `scidds` and `amounts`.
In fact you dereference it below using `*` instead of `[]`.